clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/ArraysTests.scala

```diff
@@ -0,0 +1,1040 @@
+        .foreach(_ shouldNot exist)
+
+      Seq(fingerprintingDetailMetricsSource, vcfDestination, craiDestination)
+        .foreach(_ should exist)
+
+      // We don't deliver fingerprinting metrics for now because they're based on unpublished research.
+      fingerprintingSummaryMetricsSource should exist
+      fingerprintMetricsDestination shouldNot exist
+
+      Seq(
+        (vcfDestination, vcfContents),
+        (craiDestination, vcfIndexContents),
+        (fingerprintingDetailMetricsSource, fingerprintingDetailMetricsContents),
+        (fingerprintingSummaryMetricsSource, fingerprintingSummaryMetricsContents)
+      ).foreach {
+        case (destination, contents) =>
+          destination.contentAsString should be(contents)
+      }
+      succeed
+    }
+
+    result.andThen {
+      case _ =>
+        val _ = Seq(
+          vcfSource,
+          vcfDestination,
+          vcfIndexSource,
+          craiDestination,
+          fingerprintingDetailMetricsSource,
+          alignmentMetricsDestination,
+          fingerprintingSummaryMetricsSource,
+          fingerprintMetricsDestination
+        ).map(_.delete(swallowIOExceptions = true))
+    }
+  }
+
+  it should "move the vcf and vcfIndex together in GCP" in testMove()
+
+  it should "support changing the vcf and vcfIndex basename on move" in testMove(
+    changeBasename = true
+  )
+
+  it should "not move arrays without a destination" in {
+    recoverToExceptionIf[Exception] {
+      runClient(
+        ClioCommand.moveArraysName,
+        "--location",
+        Location.GCP.entryName,
+        "--chipwell-barcode",
+        randomId,
+        "--version",
+        "123"
+      )
+    }.map {
+      _.getMessage should include("--destination")
+    }
+  }
+
+  it should "not move arrays with no registered files" in {
+    val key = ArraysKey(
+      Location.GCP,
+      Symbol(s"barcode$randomId"),
+      1
+    )
+    runUpsertArrays(key, ArraysMetadata()).flatMap { _ =>
+      recoverToExceptionIf[Exception] {
+        runClient(
+          ClioCommand.moveArraysName,
+          "--location",
+          key.location.entryName,
+          "--chipwell-barcode",
+          key.chipwellBarcode.name,
+          "--version",
+          key.version.toString,
+          "--destination",
+          "gs://some-destination/"
+        )
+      }.map {
+        _.getMessage should include("Nothing to move")
+      }
+    }
+  }
+
+  def testDelete(
+    existingNote: Option[String] = None,
+    testNonExistingFile: Boolean = false,
+    force: Boolean = false,
+    workspaceName: Option[String] = None
+  ): Future[Assertion] = {
+    val deleteNote = s"$randomId --- Deleted by the integration tests --- $randomId"
+
+    val id = randomId
+    val barcode = s"barcode$id"
+    val version = 3
+
+    val vcfContents = s"$id --- I am a vcf fated to die --- $id"
+    val vcfIndexContents = s"$id --- I am an index fated to die --- $id"
```

Comment:
We should assert on [all the files-to-be-deleted](https://github.com/broadinstitute/clio/blob/develop/clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/arrays/ArraysMetadata.scala#L48), not just the vcf.